beacon-chain/sync/pending_blocks_queue.go

```diff
@@ -153,13 +153,15 @@ func (s *Service) processPendingBlocks(ctx context.Context) error {
 			}
 
 			if err := s.validateBeaconBlock(ctx, b, blkRoot); err != nil {
-				log.Debugf("Could not validate block from slot %d: %v", b.Block().Slot(), err)
-				s.setBadBlock(ctx, blkRoot)
-				tracing.AnnotateError(span, err)
-				// In the next iteration of the queue, this block will be removed from
-				// the pending queue as it has been marked as a 'bad' block.
-				span.End()
-				continue
+				if !errors.Is(ErrOptimisticParent, err) {
```

Comment:
As it turns out, we can't remove it because this will be a real error down the line. We haven't hooked up `is_optimistic_candidate_block` function yet. See:
https://github.com/prysmaticlabs/prysm/blob/develop/beacon-chain/blockchain/optimistic_sync.go#L152

@potuz